modules/aggregation-layers/src/screen-grid-layer/screen-grid-layer.js

```diff
@@ -118,6 +88,33 @@ export default class ScreenGridLayer extends Layer {
     }
   }
 
+  renderLayers() {
+    if (!this.state.supported) {
+      return [];
+    }
+    const {weights, maxTexture, numRow, numCol, numInstances} = this.state;
+    const {updateTriggers} = this.props;
+
+    return new ScreenGridCellLayer(
+      this.props,
+      this.getSubLayerProps({
+        id: 'cell-layer',
+        updateTriggers
+      }),
+      {
+        data: {
```

Comment:
This is going to invalidate all attributes in the sub layer every time viewport changes. Cache the `data` object or supply `instanceCounts` as a top-level prop.

Reply:
updated to cache data prop.